Extensions/one_click_postage.js

```diff
@@ -381,45 +367,26 @@ XKit.extensions.one_click_postage = new Object({
 
 			XKit.svc.post.update(m_object, kitty_data.kitten)
 				.then(response => {
-					try {
-						let responseData = response.json();
-						if (responseData.errors === false) {
-							$(obj).removeClass("xkit-button-working");
-							$(obj).addClass("xkit-button-done");
-						} else {
-							this.show_error("INOCP901", state);
-							$(obj).removeClass("xkit-button-working");
-							$(obj).addClass("xkit-button-error");
-						}
-					} catch (e) {
-						this.show_error("INOCP181", state);
-						$(obj).removeClass("xkit-button-working");
-						$(obj).addClass("xkit-button-error");
+					let responseData = response.json();
+					$(obj).removeClass("xkit-button-working");
```

Comment:
let's move this into the `errors === false` conditional so it's clearer that the `catch` is going to handle the error button state transition

Reply:
actually this whole thing might work a little better as an early return.

```js
if (responseData.errors) {
  throw response;
}
```

(shouldn't this also be `error`?)

Reply:
`error` is only ever present in bad HTTP responses, good responses always have `errors` (even if it's just `errors: false`)